workspace/__ardulib__/Graphics/XGraphics.h

```diff
@@ -0,0 +1,31 @@
+
+#ifndef X_GRAPHICS_H
+#define X_GRAPHICS_H
+
+#include <Arduino.h>
+
+#include "XRender.h"
+#include "XVector2.h"
+
+using xod::XColor;
+
+class XGraphics {
+protected:
+    XGraphics* _parent;
+public:
+    XGraphics();
+
+    virtual XColor getFGColor() const;
+
+    virtual XVector2<int16_t> getCanvasPivot() const;
+    virtual int16_t getCanvasWidth() const;
+    virtual int16_t getCanvasHeight() const;
```

Comment:
Instead of keeping three virtual methods for the polymorphism which is used in a single place, I’d replace all three with a single:

```cpp
virtual const Canvas* getUpstreamCanvas() const;
```

Do not forget the forward declaration in this case:

```cpp
class Canvas;
```